### PR TITLE
A sample showing how to deploy VLAN (Interconnect) attachment

### PR DESCRIPTION
**Pull request by Dnefedkin**
vlan_attachment deployment manager sample shows how  automate creation of VLAN attachments and Cloud Routers when you already have provisioned interconnects before.

### CODE REVIEW

**Comment by Dnefedkin** [comment]
I signed it

File: examples/v2/vlan_attachment/README.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+## Deploy the template
+
+1. Change values of properties in  `vlan_attachment.yaml`, do not uncomment peer_ip_address property so far
+
+2. Deploy with the following command:
+
+    gcloud deployment-manager deployments create <your_deployment_name> --config ./vlan_attachment.yaml
+
+3. Find customer_router_ip output value by [viewing the configuration layout](https://cloud.google.com/deployment-manager/docs/deployments/viewing-manifest#configuration_layout)
```
Comment (krwenholz) [inline examples/v2/vlan_attachment/README.md]: You could also provide the gcloud command to describe the deployment and actually grab the output. I think something like `gcloud deployment-manager deployments describe <your_deployment_name> --format="yaml(outputs)"`... or something like that.

File: examples/v2/vlan_attachment/README.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+1. Change values of properties in  `vlan_attachment.yaml`, do not uncomment peer_ip_address property so far
+
+2. Deploy with the following command:
+
+    gcloud deployment-manager deployments create <your_deployment_name> --config ./vlan_attachment.yaml
+
+3. Find customer_router_ip output value by [viewing the configuration layout](https://cloud.google.com/deployment-manager/docs/deployments/viewing-manifest#configuration_layout) 
+
+4. Uncomment peer_ip_address property in `vlan_attachment.yaml` and set it's value to the IP address part of  customer_router_ip (e.g. if customer_router_ip is 169.254.68.162/29, use 169.254.68.162 as a value)
```
Comment (krwenholz) [inline examples/v2/vlan_attachment/README.md]: Can you clarify why this output can't be used directly in the deployment without an update? Is there no operation to poll on or something?

File: examples/v2/vlan_attachment/vlan_attachment.py [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+	resources = []
+
+	router_name = context.env['deployment'] + '-router'
+	vlan_attach_name = context.env['deployment'] + '-vlan-attach'
+	router_interface_name = router_name + '-if' 
+	bgp_peer_name = router_interface_name + '-peer'
+
+	# Adding Cloud Router
+	resources.append({
+		'name': router_name,
```
Comment (krwenholz) [inline examples/v2/vlan_attachment/vlan_attachment.py]: These are tabbed over pretty far. Could you use a format more like our other samples? Also, I'm not sure how to check on GitHub but please make sure you're using spaces and not tabs.

For example:

https://github.com/GoogleCloudPlatform/deploymentmanager-samples/blob/master/examples/v2/cloud_functions/python/cloud_function.py

File: examples/v2/vlan_attachment/README.md (hunk 2 above)
Comment (Dnefedkin) [reply examples/v2/vlan_attachment/README.md]: Deployment manager does not support reference post processing. 

'$(ref.' + vlan_attach_name + '.customerRouterIpAddress)' returns something like  '169.254.68.162/29'  . I need just '169.254.68.162' at the next step, but I cannot just do something like 

'peerIpAddress': extractIPAddressFromCIDR('$(ref.' + vlan_attach_name + '.customerRouterIpAddress)')
or
'peerIpAddress': '$(ref.' + vlan_attach_name + '.customerRouterIpAddress).split("/")[0]'

See internal 69658438 on this topic

File: examples/v2/vlan_attachment/vlan_attachment.py (hunk 3 above)
Comment (Dnefedkin) [reply examples/v2/vlan_attachment/vlan_attachment.py]: Fixed

File: examples/v2/vlan_attachment/README.md (hunk 1 above)
Comment (Dnefedkin) [reply examples/v2/vlan_attachment/README.md]: Added display_outputs.sh and explained how to use it in README

**Comment by Dnefedkin** [comment]
@krwenholz , thanks for your feedback and suggestions!  PTAL

File: examples/v2/vlan_attachment/README.md (hunk 2 above)
Comment (krwenholz) [reply examples/v2/vlan_attachment/README.md]: Ah okay I didn't realize that. Thanks for clarifying.

**Comment by krwenholz** [comment]
Excellent contribution. Thanks Dmitry.

**Pull request merged**